clients/admin-ui/src/features/integrations/configure-monitor/ConfigureMonitorForm.tsx

```diff
@@ -0,0 +1,167 @@
+
+  const CURRENT_DATE_ISO = formatISO(Date.now(), { representation: "date" });
+
+  const validationSchema = Yup.object().shape({
+    name: Yup.string().required().label("Name"),
+    execution_start_date: isEditing
+      ? Yup.string().nullable()
+      : Yup.date()
+          .transform((dateString) => new Date(dateString))
+          .min(CURRENT_DATE_ISO)
+          .label("Execution start date"),
+    execution_frequency: Yup.string().required().label("Execution frequency"),
+  });
+
+  const [putMonitorMutationTrigger, { isLoading }] =
+    usePutDiscoveryMonitorMutation();
+
+  const { toastResult } = useQueryResultToast({
+    defaultSuccessMsg: `Monitor ${
+      isEditing ? "updated" : "created"
+    } successfully`,
+    defaultErrorMsg: `A problem occurred while ${
+      isEditing ? "updating" : "creating"
+    } this monitor`,
+  });
+
+  const handleSubmit = async (values: FormValues) => {
+    const startDate = new Date(values.execution_start_date);
+    startDate.setUTCHours(0, 0, 0);
+
+    const payload: MonitorConfig = isEditing
+      ? {
+          ...monitor,
+          name: values.name,
+          execution_frequency: values.execution_frequency,
+        }
+      : {
+          name: values.name,
+          connection_config_key: integrationId!,
+          execution_frequency: values.execution_frequency,
+          execution_start_date: startDate.toISOString(),
+          classify_params: {
+            num_samples: 25,
+            num_threads: 1,
+          },
+        };
+
+    const result = await putMonitorMutationTrigger(payload);
+    toastResult(result);
+    if (!isErrorResult(result)) {
+      onAdvance(result.data);
+    }
+  };
+
+  const handleNextClicked = (values: FormValues, isDirty: boolean) => {
+    if (isDirty) {
+      handleSubmit(values);
+    } else {
+      onAdvance(monitor!);
+    }
+  };
+
+  const initialValues = {
+    name: monitor?.name ?? "",
+    execution_start_date: CURRENT_DATE_ISO,
+    execution_frequency: monitor?.execution_frequency,
+  };
+
+  return (
+    <Formik
+      initialValues={initialValues}
+      enableReinitialize
+      onSubmit={handleSubmit}
+      validationSchema={validationSchema}
+    >
+      {({ values, dirty, isValid, resetForm }) => (
+        <Form>
+          <VStack alignItems="start" spacing={6} mt={4}>
+            <CustomTextInput
+              name="name"
+              id="name"
+              label="Name"
+              isRequired
+              variant="stacked"
+            />
+            {!isEditing && (
+              <CustomDatePicker
+                name="execution_start_date"
+                label="Execution start date"
+                id="execution_start_date"
+                isRequired
+                // @ts-ignore - TS is looking for a number here and doesn't
+                // know the string is formatted correctly
+                minValue={CURRENT_DATE_ISO}
```

Comment:
Can we just update the interface of `<CustomDatePicker>` component to accept a string for `minValue` instead? Looks like it gets used on the `min` property of Chakra's Input component that accepts strings anyway.
```min?: string | number | undefined```